crawlers/arxiv_crawler.py

```
@@ -0,0 +1,117 @@
+            exit(1)
+
+        # setup requests session and mount adapter to retry requests
+        self.session = create_session_with_retries()
+
+        # define query for arxiv: search for the query in the "computer science" (cs.*) category
+        query = f"cat:{category}.* AND " + ' AND '.join([f'all:{q}' for q in query_terms])
+        if self.cfg.arxiv_crawler.sort_by == 'citations':
+            # for sort by n_citations We pull 100x papers so that we can get citations and enough highly cited paper.
+            search = arxiv.Search(
+                query = query,
+                max_results = n_papers*100,
+                sort_by = arxiv.SortCriterion.Relevance,
+                sort_order = arxiv.SortOrder.Descending
+            )
+        else:
+            search = arxiv.Search(
+                query = query,
+                max_results = n_papers,
+                sort_by = arxiv.SortCriterion.submittedDate,
+                sort_order = arxiv.SortOrder.Descending
+            )
+
+        # filter by publication year
+        papers = []
+        try:
+            for result in search.results():
+                date = result.published.date()
+                if date.year < year:
+                    continue
+                id = result.entry_id.split('/')[-1]
+                papers.append({
+                    'id': result.entry_id,
+                    'citation_count': self.get_citations(id),
+                    'url': result.pdf_url,
+                    'title': result.title,
+                    'authors': result.authors,
+                    'abstract': result.summary,
+                    'published': str(date)
+                })
+        except Exception as e:
+            logging.info(f"Exception {e}, we have {len(papers)} papers already, so will continue with indexing")
+
+        # sort by citation count and get top n papers
+        if self.cfg.arxiv_crawler.sort_by == 'citations':
```

Should we throw an exception if, after filtering, papers is an empty array?